src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java

```diff
@@ -664,31 +674,30 @@ private void checkGraalVMPermissions(String graalvmHome) {
     }
 
     /**
-     * Gets the Java version that GraalVM bundles
-     * @return the Java version of the GraalVM's build
+     * Get the output of running 'java -version' on the GraalVM JDK
+     * @return The output of executing 'java -version' on the GraalVM JDK
      * @throws NullPointerException when the configuration is null
      * @throws IllegalArgumentException when the configuration doesn't contain a property graalPath
      * @throws IOException if the Java version can't be found
      */
-    public Version getGraalVMJavaVersion() throws IOException {
-        ProcessRunner graalJava = null;
+    private String getJavaVersionString() throws IOException {
         try {
-            graalJava = new ProcessRunner(getGraalVMBinPath().resolve("java").toString(), "-version");
+            ProcessRunner graalJava = new ProcessRunner(getGraalVMBinPath().resolve("java").toString(), "-version");
             if (graalJava.runProcess("check version") != 0) {
                 throw new IllegalArgumentException("$GRAALVM_HOME/bin/java -version process failed");
             }
+            String response = graalJava.getResponse();
+            if (response == null || response.isEmpty()) {
+                throw new IOException("Couldn't determine GraalVM's Java version");
+            }
+            return response;
         } catch (InterruptedException e) {
             throw new IllegalArgumentException("$GRAALVM_HOME/bin/java -version process failed");
         }
-        List<String> responses = graalJava.getResponses();
-        if (responses == null || responses.isEmpty()) {
-            throw new IOException("Couldn't determine GraalVM's Java version");
-        }
-        return parseGraalVMJavaVersion(responses.get(0));
     }
 
     static Version parseGraalVMJavaVersion(String versionString) {
-        String realVersion = versionString.replaceAll("-internal", "");
+        String realVersion = versionString.replace("-internal", "");
```

Comment:
I don't know exactly what case this `-internal` string covers, but it wouldn't hurt adding a matching parser test as well.

Reply:
If you clone the JDK repo and build it yourself you get something like:
```
$ jdk/build/macosx-aarch64-server-release/images/jdk/bin/java -version
openjdk version "22-internal" 2024-03-19
OpenJDK Runtime Environment (build 22-internal-adhoc.<$user.name>.jdk)
OpenJDK 64-Bit Server VM (build 22-internal-adhoc.<$user.name>.jdk, mixed mode, sharing)
```

Reply:
@kkriske Just for my understanding, are you planning on adding more tests to the internal (and maybe the "ea" case, if you use https://jdk.java.net/22/ for instance), or is this PR ready?

Reply:
What's still left to get this merged?
Should I add these -internal and -ea version string? Those are not supported I'd think because they don't contain any GraalVM version, which resullts in a `IllegalArgumentException`.

Reply:
For me it should be ready. 

I was just asking because of your comment:
> but it wouldn't hurt adding a matching parser test as well.

We probably don't need to worry about testing those cases now. `-internal` is removed, and GraalVM doesn't use `-ea` (just tested latest nightly build from [here](https://github.com/graalvm/graalvm-ce-dev-builds/releases/)).

Reply:
I missed your last message. I'm currently not planning on adding anything more, mainly because I don't really understand the use-cases of the ea and internal versions because the version checks will reject them as they do not have GraalVM versions.

Reply:
@jperedadnr Just wanna bump this.
If nothing is still blocking this, can we get this released?